lib/addons/olm/index.ts

```diff
@@ -0,0 +1,71 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { Construct } from 'constructs';
+import { loadYaml } from "../../utils/yaml-utils";
+import { KubectlProvider, ManifestDeployment } from "../helm-addon/kubectl-provider";
+
+const OLMDownloadURL = "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/";
+
+export interface OlmAddOnProps {
+  /**
+  * OLM version
+  */
+  version?: string
+}
+
+/**
+ * Default OLM properties
+ */
+const defaultProps = {
+  version: "v0.21.2"
+};
+
+export class OlmAddOn implements ClusterAddOn {
+    readonly props: OlmAddOnProps;
+    readonly manifestUrls: string[];
+
+    constructor(props?: OlmAddOnProps) {
+      this.props = {...defaultProps, ...props};
+      this.manifestUrls = [
+        OLMDownloadURL+this.props.version+"/crds.yaml",
+        OLMDownloadURL+this.props.version+"/olm.yaml"
+      ];
+    }
+
+    deploy(clusterInfo: ClusterInfo): void | Promise<Construct> {
+      let previousResource: Construct;
+      const resources: Construct[] = [];
+      this.manifestUrls!.map((manifestUrl, urlIndex) => {
+        /* eslint-disable */
+        const request = require('sync-request');
+        const response = request('GET', manifestUrl);
+        // workaround: KubectlProvider is based on Lambda, which fails if the manifests are too large, so we:
+        // 1. break the YAMLs
+        // 2. remove descriptions
+        const doc: string = response.getBody().toString();
+        const trimmedDoc = doc.replace(/description:.*[a-zA-Z0-9].*\n/g, 'description: removed\n');
+        const manifests = trimmedDoc.split("---").filter(e=>e.length>0).map(e => loadYaml(e));
+
+        const batchResources = manifests.map((manifest, docIndex) => {
+          const manifestDeployment: ManifestDeployment = {
+            name: "olmIdx"+urlIndex+docIndex,
+            namespace: "olm",
+            manifest: [manifest],
```

Comment:
why manifest is an array here? should be an actual doc.

Reply:
addManifest finally the KubernetesManifest and it looks to me "manifest" is an array type in the param list, as per here: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks.KubernetesManifest.html#construct-props ?

Reply:
nm, I think we should do a better job on reflecting the types.